### DISCUSSION TITLE
How to change appium UI's log file location?

### DISCUSSION THREAD

**Discussion by Jiandang-Kent** [appium/appium#15913] [Q&A]
Had changed Appium's configuration in the advanced tab by changing server address to 127.0.0.1 and log file location to ~/path/to/appium.log on my Mac. But it seems that the configuration doesn't work. The log file is still in a random directory with name 'appium-log-files.txt'.

**Comment by portyu9** [comment]
Hi @Jiandang-Kent 

Use an **absolute path** instead of:

```text
~/path/to/appium.log
```

For example on macOS:

```text
/Users/yourusername/path/to/appium.log
```

The most reliable approach is to start Appium directly with:

```bash
appium --log /Users/yourusername/path/to/appium.log
```

or:

```bash
appium -g /Users/yourusername/path/to/appium.log
```

`~` is normally expanded by your shell, so a GUI-launched Appium process may not resolve it as expected.

Also note that Appium Desktop has been deprecated; with current Appium, run the Appium server separately from the CLI and use Appium Inspector as the GUI client.

So the fix is: **use an absolute path and configure the server's `--log` option directly rather than relying on the old Desktop UI log-path setting.**

Please mark this answer as accepted if it helped, thank you.
